squidDrone/src/com/com_interface.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef SRC_COM_INTERFACE_HPP_
+#define SRC_COM_INTERFACE_HPP_
+
+#ifndef UNIT_TEST
+#include "com_message_buffer.hpp"
+#else
+#include "com_message_buffer_mock.hpp"
+#endif
+
+#include <array>
+
+namespace com {
+
+  /**
+   * @class ComInterface 
+   * @brief This is a purely virtual class to act as
+   * a facade for the device specific implmentation.
+   * 
+   */
+  class ComInterface {
+    public:
+      /** 
+       * @brief Default constructor may not be used.
+       * 
+       */
+      ComInterface() = delete;
+
+      /** 
+       * @brief Construct a new Com Interface object
+       * 
+       */
+      explicit ComInterface(std::unique_ptr<com::ComMessageBuffer> msg_buf) : msg_buffer(std::move(msg_buf)){};
+
+      /**
+       * @brief Virtual destruktor. 
+       *
+       */
+      virtual ~ComInterface() = default;
+
+      /** 
+       * @brief Get the data packet object
+       * 
+       * @return decltype(array<std::uint8_t, 32>) 
```

Comment:
return documentation says decltype(array<std::uint8_t, 32>) but actually is std::array<std::uint8_t, 32>

Reply:
This is autogenerated by the doxygen documentation generator.

Reply:
Accepted it is technically not wrong, but a odd way to write it.